extern/noah-mp-modular/CMakeLists.txt

```diff
@@ -0,0 +1,71 @@
+cmake_minimum_required(VERSION 3.12)
+enable_language( Fortran )
+add_subdirectory(../iso_c_fortran_bmi ${CMAKE_BINARY_DIR}/iso_c_bmi)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/../../cmake/")
+# Uncomment this and rebuild artifacts to enable debugging
+#set(CMAKE_BUILD_TYPE Debug)
+set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -g -fbacktrace -fbounds-check -Wall")
+project(noah-mp-modular VERSION 1.0.0 DESCRIPTION "External BMI Models Shared Libraries")
+
+set( CMAKE_Fortran_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/fortran)
+
+#### Add variables for individual libraries that are used withing the *.pc.in files
+# Surface 
+set(SURFACE_LIB_NAME_CMAKE surfacebmi)
+set(SURFACE_LIB_DESC_CMAKE "External Noah-MP Surface Shared Library")
+
+# Make sure these are compiled with this directive
+add_compile_definitions(BMI_ACTIVE)
+find_package (NetCDF REQUIRED COMPONENTS F90)
+
+set(MODEL_SOURCE_DIR noah-mp-modular/modules/surface_bmi/src/)
+set(BMI_SOURCE_DIR noah-mp-modular/modules/surface_bmi/bmi/)
+
+file(GLOB BMI_SOURCE ${BMI_SOURCE_DIR}*.f90)
+file( GLOB MODEL_SOURCES ${MODEL_SOURCE_DIR}*.f90)
+
+list(APPEND MODEL_SOURCES noah-mp-modular/modules/surface_bmi/driver/noahmp_ascii_read.f90
+                          noah-mp-modular/modules/surface_bmi/driver/noahmp_output.f90)
+
+add_library(surfacebmi_obj OBJECT ${MODEL_SOURCES} )
+target_compile_options(surfacebmi_obj PUBLIC -cpp -ffree-line-length-none)
+target_include_directories(surfacebmi_obj PUBLIC noah-mp-modular/modules/surface_bmi/driver)
+
+#HACK FIXME RM
+set(NETCDF_F90_INCLUDE_DIRS /usr/include/openmpi-x86_64/)
```

Comment:
we should at the very least try to read this variable from the shell before using a hard coded value (which is wrong for many distributions)

Reply:
This snuck in, I'll remove it.  The find netcdf cmake function doesn't seem to work correctly on some systems, so this was hard coded in for that specific instance.

Reply:
This is turning out to be tricky getting NetCDF to link correctly in a generic cmake file.